src/toil/batchSystems/singleMachine.py

```diff
@@ -570,31 +569,6 @@ def _handleChild(self, pid: int) -> None:
             # If we killed it then it shouldn't show up in the queue.
             self.outputQueue.put(UpdatedBatchJobInfo(jobID=jobID, exitStatus=statusCode, wallTime=time.time() - info.time, exitReason=None))
 
-        # Make absolutely sure all processes in the group have received their
```

Comment:
We might still want this loop so users know that there are child processes that can't be reaped, if they are not running on a toil cluster?

Reply:
I agree! I will bring the loop back and update the comment accordingly.

Reply:
> @DailyDreaming If we keep the loop, and we run inside someone else's container, running their user code that leaves orphan zombie child processes, we will get stuck.

I updated so that we'll just leave a warning message if there are zombie processes left behind instead, since the loop wouldn't actually do anything except blocking the workflow from completion. Please let me know what you think though.